### PR TITLE
add person, student, and teacher classes

### PR DESCRIPTION
**Pull request by ginabeki**
### Summary

> In this project, I did the following:

- Implemented classes for person, student, and teacher.
- Implement encapsulation and inheritance with Ruby.
- Used IRB to validate my result
- Used linters to clear errors

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(classroom)
```
Comment (tresorsawasawa) [inline student.rb:4]: - [ ] Good job Inheriting the `Student` class from `Person`. But you should ensure that the `Constructor` extends parent's constructor by adding `@specialization `and a `parameter` for it i.e.:
```ruby
  ...
  def initialize(age, classroom, name = "Unknown", parent_permission = true)
    ...
  end
  ...
```

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'Person'
+
+class Teacher < Person
+  def initialize(specialization)
```
Comment (tresorsawasawa) [inline teacher.rb:4]: - [ ] Good job Inheriting the `Teacher` class from `Person`. As I mentioned in the above comment, you should ensure that the `Constructor` extends parent's constructor by adding `@specialization `and a `parameter` for it i.e.:
```ruby
  ...
  def initialize(age, specialization, name = "Unknown", parent_permission = true)
    ...
  end
...
```

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @ginabeki 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OLIPLICHE — APPROVED** [review]
Hi @ginabeki, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the change requested by the review👏 👏
- Linter is green (the last commit has a green check mark (✅)) 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**